File: examples/nodejs/10_pubsub.js
const Redis = require("ioredis");

async function run() {
    console.log("--- 10 Pub/Sub ---");

    const sub = new Redis({ host: "localhost", port: 6379 });
    const pub = new Redis({ host: "localhost", port: 6379 });

    await sub.subscribe("node_channel");
    console.log("Subscribed to 'node_channel'");

    sub.on("message", (channel, message) => {
        console.log(`[Sub] Received on ${channel}: ${message}`);
        if (message === "QUIT") {
            sub.disconnect();
            pub.disconnect();
        }
    });

    // Small delay to ensure subscription is active
    await new Promise(r => setTimeout(r, 100));

    // Publish
    console.log("[Pub] Sending Hello...");
    await pub.publish("node_channel", "Hello from Node.js");

    await new Promise(r => setTimeout(r, 100));
    console.log("[Pub] Sending QUIT...");
    await pub.publish("node_channel", "QUIT");
}

run().catch(console.error);

File: examples/vbnet/10_PubSub.vb
Imports StackExchange.Redis
Imports System
Imports System.Threading

Module Program
    Sub Main()
        Dim redis = ConnectionMultiplexer.Connect("localhost")
        Dim sub = redis.GetSubscriber()

        Console.WriteLine("--- 10 Pub/Sub ---")

        sub.Subscribe("vb_channel", Sub(channel, message)
            Console.WriteLine($"[Sub] Received: {message}")
        End Sub)

        Console.WriteLine("Subscribed. Sending in 1s...")
        Thread.Sleep(1000)

        sub.Publish("vb_channel", "Hello from VB.NET")
        Thread.Sleep(500)

        sub.UnsubscribeAll()
        redis.Close()
    End Sub
End Module
